File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        a,b,c = 0,1,2
        ans =0
        flag =  False
        while c < len(num):
            if num[a] == num[b] and num[a] == num[c]:
                flag = True
                ans = max(ans,int(num[a]))
            a,b,c = b,c,c+1
        return str(ans)*3 if flag else ""

File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        char ans = 0;

        for (int i = 0; i + 2 < num.size(); i++) {
            if (num[i] == num[i + 1] && num[i] == num[i + 2]) {
                ans = max(ans, num[i]);
            }
        }

        if (ans == 0)
            return "";

        return string(3, ans);
    }
};